test/unit_test/common/test_float_utils.py

```
@@ -86,3 +89,69 @@ def test_whitespace_strings(self):
         result = get_float("  invalid  ")
         assert math.isinf(result)
         assert result < 0
+
+
+class TestNormalizeOverlappedPercent:
+    """Round-vs-truncate fix for #17418.
+
+    The function used ``int()`` after multiplying a [0,1) fraction by 100,
+    which truncates IEEE-754 ``×100`` results that land just below an
```

Ruff (0.15.21)

[warning] Docstring contains ambiguous `×` (MULTIPLICATION SIGN). Did you mean `x` (LATIN SMALL LETTER X)?

(RUF002)